lib/github/kv.rb

```diff
@@ -270,6 +270,30 @@ def mdel(keys)
       nil
     end
 
+    # ttl :: String -> Result<[Time | nil]>
+    #
+    # Returns the expires_at time for the specified key or nil.
+    #
+    # Example:
+    #
+    #  kv.ttl("foo")
+    #    # => #<Result value: 2018-04-23 11:34:54 +0200>
+    #
+    #  kv.ttl("foo")
+    #    # => #<Result value: nil>
+    #
+    def ttl(key)
+      validate_key(key)
+
+      Result.new {
+        kvs = GitHub::SQL.results(<<-SQL, :keys => [key], :connection => connection).to_h
+          SELECT `key`, expires_at FROM key_values WHERE `key` IN :keys AND (`expires_at` IS NULL OR `expires_at` > NOW())
```

Comment:
Minor nit: this could probably be a wee bit more simple with a where `key = ...` instead of the `IN ...`. Might also be able to get away with only selecting `expires_at` and using `GitHub::SQL.value` to avoid a hash and hash lookup.